### ISSUE TITLE
dxc compiles invalid program

### ISSUE THREAD

**Issue by dmpots** [Microsoft/DirectXShaderCompiler#182]
The following program has a syntax error, but dxc compiles it anyway (it will assert in a debug build). It is missing an `=` after the `A[]` part.

```
float main(float i : I) : SV_Target {
    float A[] {
        1, 2, 3
    };
    return A[i];
}
```

FXC reports an error:
```
error X3000: syntax error: unexpected integer constant
```

**Comment by youngkim93** [comment]
This should be fixed with #404. The compiler was getting confused with legacy effect state syntax with braces with c++ style aggregate initialization. On our parser level we are ignoring everything that's in the braces if it's considered effect state syntax. To use braces as an aggregate initialization we are requiring them to use equal sign('='), which is different from C++11 syntax (http://en.cppreference.com/w/cpp/language/aggregate_initialization).

**Comment by marcelolr** [comment]
@dmpots please open a new issue if this is still causing trouble in debug or you think users are going to get confused by this behavior.
